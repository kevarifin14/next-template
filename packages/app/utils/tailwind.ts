export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};
