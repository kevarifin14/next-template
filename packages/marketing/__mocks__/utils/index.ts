import { HiOutlineDesktopComputer, HiOutlineGlobe, HiOutlineLightningBolt } from 'react-icons/hi';

export const mockEmail = 'mock@gmail.com';

export const mockTitle = 'mockTitle';
export const mockDescription = 'mockDescription';
export const mockSrc = 'mockSrc';

export const mockFeatures = [
  {
    name: 'Marketing',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineGlobe,
  },
  {
    name: 'Application',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineDesktopComputer,
  },
  {
    name: 'Design System',
    description: 'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.',
    icon: HiOutlineLightningBolt,
  },
];
