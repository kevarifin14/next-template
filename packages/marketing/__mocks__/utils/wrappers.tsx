import { NotificationsProvider, AppearanceProvider } from 'shared';

export const notificationsProviderWrapper = ({ children }) => (
  <NotificationsProvider>{children}</NotificationsProvider>
);

export const appearanceProviderWrapper = ({ children }) => (
  <AppearanceProvider>{children}</AppearanceProvider>
);
