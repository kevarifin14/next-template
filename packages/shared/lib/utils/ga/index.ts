export const pageview = (url) => {
  window.gtag("config", process.env.NEXT_PUBLIC_GOOGLE_ANALYTICS, {
    page_path: url,
  });
};

type EventCategory =
  | "lead"
  | "outbound link"
  | "call to action"
  | "login"
  | "swap";

export const event = (
  category: EventCategory,
  action,
  label = "",
  params = {}
) => {
  if (process.env.NODE_ENV === "production") {
    window.gtag("event", action, {
      event_category: category,
      event_label: label,
      ...params,
    });
  }
};
