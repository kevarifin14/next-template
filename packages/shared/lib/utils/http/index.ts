import QueryString from 'query-string';

export function generateQueryString(params) {
  return QueryString.stringify(params);
}

export function generateEndpoint(endpoint, params) {
  if (Object.keys(params).length === 0) {
    return endpoint;
  }
  return `${endpoint}?${generateQueryString(params)}`;
}

export const get = async (endpoint, params = {}) => {
  const endpointWithParams = generateEndpoint(endpoint, params);
  const response = await fetch(endpointWithParams);

  if (response.status === 200) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};

export const post = async (endpoint, body, headers?) => {
  const response = await fetch(
    endpoint,
    {
      method: 'POST',
      body: JSON.stringify(body),
      headers: {
        'Content-Type': 'application/json',
        ...headers,
      },
    },
  );

  if (response.status >= 200 && response.status < 400) {
    const responseJson = response.json();
    return responseJson;
  }
  throw new Error(`${response.status}`);
};
