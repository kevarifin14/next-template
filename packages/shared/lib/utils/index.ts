export function splitName(name: string) {
  return {
    firstName: name.split(" ").slice(0, -1).join(" "),
    lastName: name.split(" ").slice(-1).join(" "),
  };
}
