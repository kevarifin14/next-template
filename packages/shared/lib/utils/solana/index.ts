/* istanbul ignore file */
import {
  AccountLayout,
  u64,
  AccountInfo as TokenAccountInfo,
  MintInfo,
} from "@solana/spl-token";
import { PublicKey } from "@solana/web3.js";

export const deserializeTokenAccount = (address: PublicKey, data: Buffer) => {
  const {
    mint,
    owner,
    amount,
    delegateOption,
    delegate,
    delegatedAmount,
    state,
    isNative,
    isNativeOption,
    closeAuthority,
    closeAuthorityOption,
  } = AccountLayout.decode(data);

  const accountInfo: TokenAccountInfo = {
    address,
    mint: new PublicKey(mint),
    owner: new PublicKey(owner),
    amount: u64.fromBuffer(amount),
    isInitialized: state !== 0,
    isFrozen: state === 2,

    ...(delegateOption === 0
      ? { delegate: null, delegatedAmount: new u64(0) }
      : {
          delegate: new PublicKey(delegate),
          delegatedAmount: u64.fromBuffer(delegatedAmount),
        }),

    ...(isNativeOption === 1
      ? { rentExemptReserve: u64.fromBuffer(isNative), isNative: true }
      : { rentExemptReserve: null, isNative: false }),

    ...(closeAuthorityOption === 0
      ? { closeAuthority: null }
      : { closeAuthority: new PublicKey(closeAuthority) }),
  };

  return accountInfo;
};

export const diffTokenAccountInfoAmounts = (
  prevTokenAccountInfo: TokenAccountInfo,
  tokenAccountInfo: TokenAccountInfo
) => {
  if (!prevTokenAccountInfo) {
    return tokenAccountInfo.amount;
  }

  return tokenAccountInfo.amount.sub(prevTokenAccountInfo.amount);
};

export const fromLamports = (
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo
) => {
  if (!tokenAccountInfo) {
    return 0;
  }

  const amount =
    typeof tokenAccountInfo === "number"
      ? tokenAccountInfo
      : tokenAccountInfo.amount.toNumber();

  const precision = 10 ** (mintInfo?.decimals || 9);
  return amount / precision;
};

export function formatTokenAmount(
  tokenAccountInfo: TokenAccountInfo | number,
  mintInfo?: MintInfo,
  prefix = "",
  suffix = "",
  precision = 3
): string {
  return `${[prefix]}${fromLamports(tokenAccountInfo, mintInfo).toFixed(
    precision
  )}${suffix}`;
}

export const formatDollar = (value) => {
  const formatter = new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
  });

  return formatter.format(value);
};
