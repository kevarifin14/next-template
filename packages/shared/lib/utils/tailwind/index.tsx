import { ComponentProps } from 'react';

export type TailwindSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl';

export const classNames = (...classes) => classes.filter(Boolean).join(' ');

export const sizeToDimension = {
  xs: 6,
  sm: 8,
  md: 10,
  lg: 12,
  xl: 16,
  '2xl': 20,
  '3xl': 24,
  '4xl': 28,
};

export const sizeToClassName = (size: TailwindSize) => {
  const dimension = sizeToDimension[size];
  return `h-${dimension} w-${dimension}`;
};

export type AlertType = 'success' | 'info' | 'error' | 'warning';

export type Heroicon = (props: ComponentProps<'svg'>) => JSX.Element; // eslint-disable-line