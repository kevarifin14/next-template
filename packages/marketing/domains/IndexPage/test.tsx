import { render } from '@testing-library/react';

import { notificationsProviderWrapper } from '__mocks__/utils/wrappers';

import { IndexPage } from '.';

describe('IndexPage', () => {
  const renderIndexPage = (props = {}) => (
    render(
      <IndexPage
        {...props}
      />,
      { wrapper: notificationsProviderWrapper },
    )
  );

  it('matches snapshot', () => {
    const { baseElement } = renderIndexPage();
    expect(baseElement).toMatchSnapshot();
  });
});
