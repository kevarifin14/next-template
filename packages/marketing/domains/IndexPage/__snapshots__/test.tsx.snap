// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage matches snapshot 1`] = `
<body>
  <div>
    <div
      class="fixed max-w-sm w-full right-0 top-0 flex flex-col justify-end items-start space-y-4 pointer-events-none z-30 p-4"
    />
    <div>
      <div
        class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-none"
      >
        <div
          class="max-w-7xl mx-auto text-center prose-xl dark:prose-light py-12"
        >
          <h1
            class="font-extrabold m-0"
          >
            <span
              class="mr-3"
            >
              The Next Template to Build
            </span>
            <span
              class="text-primary"
            >
              Anything
            </span>
          </h1>
          <p
            class="mt-3"
          >
            Stop writing boilerplate code and start building from day one.
          </p>
          <form
            class="flex space-x-4 max-w-lg mx-auto py-8"
          >
            <div
              class="w-full"
            >
              <div
                class="relative"
              >
                <input
                  class="shadow-sm focus:outline-none w-full border bg-light border-light-400 rounded-md block text-gray-900 placeholder-gray-500 focus:border-primary-600 focus:ring-primary-600 focus:ring-1 dark:bg-dark-light dark:border-transparent dark:text-white px-4 py-2"
                  id="123456879"
                  name="email"
                  placeholder="Enter Your Email"
                  type="email"
                />
              </div>
            </div>
            <div
              class="flex-shrink-0"
            >
              <button
                class="inline-flex items-center justify-center focus:outline-none font-medium rounded-md text-white bg-gradient-to-tr from-primary-500 to-primary-800    px-4 py-2 "
                type="submit"
              >
                Get Early Access
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-7xl lg:grid lg:grid-cols-2 lg:gap-8 lg:items-center"
      >
        <div
          class="col-start-2"
        >
          <h1
            class="font-extrabold tracking-tight m-0 mb-2"
          >
            Everything out of the box
          </h1>
          <p
            class="m-0"
          >
            A landing page, application, and design system.
          </p>
          <dl
            class="mt-10 space-y-10"
          >
            <div
              class="relative"
            >
              <dt>
                <div
                  class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.055 11H5a2 2 0 012 2v1a2 2 0 002 2 2 2 0 012 2v2.945M8 3.935V5.5A2.5 2.5 0 0010.5 8h.5a2 2 0 012 2 2 2 0 104 0 2 2 0 012-2h1.064M15 20.488V18a2 2 0 012-2h3.064M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
                <h3
                  class="ml-16 m-0"
                >
                  Marketing
                </h3>
              </dt>
              <p
                class="mt-2 ml-16"
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
              </p>
            </div>
            <div
              class="relative"
            >
              <dt>
                <div
                  class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9.75 17L9 20l-1 1h8l-1-1-.75-3M3 13h18M5 17h14a2 2 0 002-2V5a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
                <h3
                  class="ml-16 m-0"
                >
                  Application
                </h3>
              </dt>
              <p
                class="mt-2 ml-16"
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
              </p>
            </div>
            <div
              class="relative"
            >
              <dt>
                <div
                  class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13 10V3L4 14h7v7l9-11h-7z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
                <h3
                  class="ml-16 m-0"
                >
                  Design System
                </h3>
              </dt>
              <p
                class="mt-2 ml-16"
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
              </p>
            </div>
          </dl>
        </div>
        <div
          class="col-start-1 row-start-1"
        >
          <img
            alt=""
            class="relative mx-auto mt-16 lg:mt-0"
            src="https://tailwindui.com/img/features/feature-example-2.png"
            width="490"
          />
        </div>
      </div>
      <div
        class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-7xl lg:grid lg:grid-cols-2 lg:gap-8 lg:items-center"
      >
        <div
          class=""
        >
          <h1
            class="font-extrabold tracking-tight m-0 mb-2"
          >
            Everything out of the box
          </h1>
          <p
            class="m-0"
          >
            A landing page, application, and design system.
          </p>
          <dl
            class="mt-10 space-y-10"
          >
            <div
              class="relative"
            >
              <dt>
                <div
                  class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.055 11H5a2 2 0 012 2v1a2 2 0 002 2 2 2 0 012 2v2.945M8 3.935V5.5A2.5 2.5 0 0010.5 8h.5a2 2 0 012 2 2 2 0 104 0 2 2 0 012-2h1.064M15 20.488V18a2 2 0 012-2h3.064M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
                <h3
                  class="ml-16 m-0"
                >
                  Marketing
                </h3>
              </dt>
              <p
                class="mt-2 ml-16"
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
              </p>
            </div>
            <div
              class="relative"
            >
              <dt>
                <div
                  class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9.75 17L9 20l-1 1h8l-1-1-.75-3M3 13h18M5 17h14a2 2 0 002-2V5a2 2 0 00-2-2H5a2 2 0 00-2 2v10a2 2 0 002 2z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
                <h3
                  class="ml-16 m-0"
                >
                  Application
                </h3>
              </dt>
              <p
                class="mt-2 ml-16"
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
              </p>
            </div>
            <div
              class="relative"
            >
              <dt>
                <div
                  class="absolute flex items-center justify-center h-12 w-12 rounded-md bg-primary text-white"
                >
                  <svg
                    class="h-6 w-6"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13 10V3L4 14h7v7l9-11h-7z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                    />
                  </svg>
                </div>
                <h3
                  class="ml-16 m-0"
                >
                  Design System
                </h3>
              </dt>
              <p
                class="mt-2 ml-16"
              >
                Lorem ipsum, dolor sit amet consectetur adipisicing elit. Maiores impedit perferendis suscipit eaque, iste dolor cupiditate blanditiis ratione.
              </p>
            </div>
          </dl>
        </div>
        <div
          class=""
        >
          <img
            alt=""
            class="relative mx-auto mt-16 lg:mt-0"
            src="https://tailwindui.com/img/features/feature-example-2.png"
            width="490"
          />
        </div>
      </div>
      <div
        class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-7xl"
      >
        <div
          class="max-w-3xl mx-auto text-center"
        >
          <h2
            class="text-3xl font-extrabold text-gray-900"
          >
            All-in-one platform
          </h2>
          <p
            class="mt-4 text-xl"
          >
            Ac euismod vel sit maecenas id pellentesque eu sed consectetur. Malesuada adipiscing sagittis vel nulla nec.
          </p>
        </div>
        <dl
          class="mt-12 space-y-10 sm:space-y-0 sm:grid sm:grid-cols-2 sm:gap-x-6 sm:gap-y-12 lg:grid-cols-4 lg:gap-x-8"
        >
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                Invite team members
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Tempor tellus in aliquet eu et sit nulla tellus. Suspendisse est, molestie blandit quis ac. Lacus.
            </dd>
          </div>
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                Notifications
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Ornare donec rhoncus vitae nisl velit, neque, mauris dictum duis. Nibh urna non parturient.
            </dd>
          </div>
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                List view
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Etiam cras augue ornare pretium sit malesuada morbi orci, venenatis. Dictum lacus.
            </dd>
          </div>
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                Boards
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Interdum quam pulvinar turpis tortor, egestas quis diam amet, natoque. Mauris sagittis.
            </dd>
          </div>
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                Keyboard shortcuts
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Ullamcorper in ipsum ac feugiat. Senectus at aliquam vulputate mollis nec. In at risus odio.
            </dd>
          </div>
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                Reporting
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Magna a vel sagittis aliquam eu amet. Et lorem auctor quam nunc odio. Sed bibendum.
            </dd>
          </div>
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                Calendars
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Sed mi, dapibus turpis orci posuere integer. A porta viverra posuere adipiscing turpis.
            </dd>
          </div>
          <div
            class="relative"
          >
            <dt
              class="flex space-x-2"
            >
              <svg
                class="text-green-500 h-6 w-6"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
                  fill-rule="evenodd"
                />
              </svg>
              <h3
                class="text-xl font-medium m-0"
              >
                Mobile app
              </h3>
            </dt>
            <dd
              class="mt-2 ml-8 text-base"
            >
              Quisque sapien nunc nisl eros. Facilisis sagittis maecenas id dignissim tristique proin sed.
            </dd>
          </div>
        </dl>
      </div>
      <div
        class="py-8 px-4 lg:py-12 lg:px-6 w-full prose dark:prose-light mx-auto max-w-7xl"
      >
        <div
          class="px-6 py-8 rounded-3xl shadow-xl dark:shadow-none bg-primary-dark grid grid-cols-1 gap-8 lg:grid-cols-2 items-center py-8 px-4 lg:px-16 lg:py-24"
        >
          <div
            class="space-y-2"
          >
            <h1
              class="text-white text-2xl font-extrabold m-0"
            >
              Want product news and updates?
            </h1>
            <p
              class="text-gray-200 m-0"
            >
              Sign up for our newsletter to stay up to date.
            </p>
          </div>
          <form
            class="flex space-x-4"
          >
            <div
              class="w-full"
            >
              <div
                class="relative"
              >
                <input
                  class="shadow-sm focus:outline-none w-full border bg-light border-light-400 rounded-md block text-gray-900 placeholder-gray-500 focus:border-primary-600 focus:ring-primary-600 focus:ring-1 dark:bg-dark-light dark:border-transparent dark:text-white px-5 py-3 text-lg"
                  id="123456879"
                  name="email"
                  placeholder="Enter Your Email"
                  type="email"
                />
              </div>
            </div>
            <div
              class="flex-shrink-0"
            >
              <button
                class="inline-flex items-center justify-center focus:outline-none font-medium rounded-md text-white bg-gradient-to-tr from-primary-500 to-primary-800    px-5 py-3 text-lg "
                type="submit"
              >
                Get Early Access
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</body>
`;
